esphome/components/dashboard_import/__init__.py

```diff
@@ -105,7 +106,17 @@ def import_config(
             raise ValueError(f"Error while fetching {url}: {e}") from e
 
         contents = req.text
+        _, ind, bsi = load_yaml_guess_indent(contents)
+
         yaml = YAML()
+        yaml.preserve_quotes = True
+        yaml.width = (
+            max(
+                (len(contents_len) for contents_len in contents.splitlines()), default=0
+            )
+            + 10
+        )
+        yaml.indent(mapping=ind, sequence=ind, offset=bsi)
```

Comment:
Instead of guessing, why not just use a good / clear indentation pattern if we are going to potentially change the incoming file anyway?
```suggestion
        yaml.indent(mapping=2, sequence=4, offset=2)
```

Reply:
I did verify that it isn't actually breaking as-is, but it does create some unexpected results -- in my use case, I have a bunch of packages listed out, some are commented out and others aren't.  The commented out ones are preserved, but the others aren't, which leaves them with inconsistent indentation (so you can't simply uncomment them, you need to figure out what spacing is in the file now -- which isn't hard, but is unexpected).  Guessing probably gives us the best chance of getting it right most of the time.